edb/server/dbview/dbview.pyx

```diff
@@ -230,15 +230,17 @@ cdef class Database:
         self._sql_to_compiled.clear()
         self._index.invalidate_caches()
 
-    cdef _cache_compiled_query(self, key, compiled: dbstate.QueryUnitGroup):
+    cdef _cache_compiled_query(
+        self, key, compiled: dbstate.QueryUnitGroup, int dbver
+    ):
         assert compiled.cacheable
 
-        existing, dbver = self._eql_to_compiled.get(key, DICTDEFAULT)
-        if existing is not None and dbver == self.dbver:
+        existing, existing_dbver = self._eql_to_compiled.get(key, DICTDEFAULT)
+        if existing is not None and existing_dbver == self.dbver:
```

Comment:
is it right to be checking against `self.dbver` but populating it with `dbver`?

Reply:
(yes, I think, right?)

Reply:
Right, `dbver` is the only guaranteed version that the current query unit was compiled upon. I'll add a comment.